saw-remote-api/python/saw_client/__init__.py

```diff
@@ -370,6 +370,132 @@ def create_ghost_variable(name: str) -> llvm.GhostVariable:
     return llvm.GhostVariable(name, server_name)
 
 
+@dataclass
+class JVMClass:
+    class_name: str
+    server_name: str
+
+
+def jvm_load_class(class_name: str) -> JVMClass:
+    name = __fresh_server_name(class_name)
+    __get_designated_connection().jvm_load_class(name, class_name).result()
+    return JVMClass(class_name, name)
+
+# TODO: this is almost identical to llvm_assume. Can we reduce duplication?
+def jvm_assume(cls: JVMClass,
+               method_name: str,
+               contract: llvm.Contract,
+               lemma_name_hint: Optional[str] = None) -> VerificationResult:
+    """Assume that the given method satisfies the given contract. Returns an
+    override linking the method and contract that can be passed as an
+    argument in calls to `jvm_verify`
+    """
+    if lemma_name_hint is None:
+        lemma_name_hint = contract.__class__.__name__ + "_" + method_name
+    name = __fresh_server_name(lemma_name_hint)
+
+    result: VerificationResult
+    try:
+        conn = __get_designated_connection()
+        res = conn.jvm_assume(cls.class_name,
+                              method_name,
+                              contract.to_json(),
+                              name)
+        result = AssumptionSucceeded(server_name=name,
+                                     contract=contract,
+                                     stdout=res.stdout(),
+                                     stderr=res.stderr())
+        __global_success = True
+        # If something stopped us from even **assuming**...
+    except exceptions.VerificationError as err:
+        __global_success = False
+        result = AssumptionFailed(server_name=name,
+                                  assumptions=[],
+                                  contract=contract,
+                                  exception=err)
+    except Exception as err:
+        __global_success = False
+        for view in __designated_views:
+            view.on_python_exception(err)
+        raise err from None
+
+    return result
+
+# TODO: this is almost identical to llvm_verify. Can we reduce duplication?
+def jvm_verify(cls: JVMClass,
+               method_name: str,
+               contract: llvm.Contract,
+               lemmas: Optional[List[VerificationResult]] = None,
```

Comment:
An optional list is a little funny, given that we treat `None` and `[]` the same. I suppose it's not entirely clear to me if this is a feature or a bug (in both this and `llvm_verify`), hah. I guess leaving it is fine...?

Reply:
Yeah, that's a good question. I didn't write the original version, so I don't know if there was a motivation behind it? The fact that there's a later check for `None` that sets lemmas to `[]` makes it seem eve less necessary.

Reply:
I'm happy to modify both this instance and the one in `llvm_verify` if you don't know of any reason to keep it this way.

Reply:
I don't have a strong opinion here - I don't think we strictly need to support both, and it does seem like just using `List` could simplify the code slightly. Your call.

Reply:
Done.